src/common/LoginConfig.ts

```diff
@@ -9,78 +9,81 @@ export class LoginConfig {
-            this.enableOIDC = false;
-            this.servicePrincipalId = secrets.getSecret("$.clientId", true);
-            this.servicePrincipalKey = secrets.getSecret("$.clientSecret", true);
-            this.tenantId = secrets.getSecret("$.tenantId", true);
-            this.subscriptionId = secrets.getSecret("$.subscriptionId", true);
+            core.debug('Reading creds in JSON...');
+            this.servicePrincipalId = this.servicePrincipalId ? this.servicePrincipalId : secrets.getSecret("$.clientId", false);
+            this.servicePrincipalKey = secrets.getSecret("$.clientSecret", false);
+            this.tenantId = this.tenantId ? this.tenantId : secrets.getSecret("$.tenantId", false);
+            this.subscriptionId = this.subscriptionId ? this.subscriptionId : secrets.getSecret("$.subscriptionId", false);
             this.resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
         }
-        this.getFederatedTokenIfNecessary();
+
+        this.audience = core.getInput('audience', { required: false });
+        this.federatedToken = null;
+
+        this.mask(this.servicePrincipalId);
+        this.mask(this.servicePrincipalKey);
     }
 
-    async getFederatedTokenIfNecessary() {
-        if (!this.enableOIDC) {
-            return;
-        }
+    async getFederatedToken() {
         try {
             this.federatedToken = await core.getIDToken(this.audience);
+            this.mask(this.federatedToken);
         }
         catch (error) {
             core.error(`Please make sure to give write permissions to id-token in the workflow.`);
             throw error;
         }
-        if (!!this.federatedToken) {
-            let [issuer, subjectClaim] = await jwtParser(this.federatedToken);
-            console.log("Federated token details: \n issuer - " + issuer + " \n subject claim - " + subjectClaim);
-        }
-        else {
-            throw new Error("Failed to fetch federated token.");
-        }
+        let [issuer, subjectClaim] = await jwtParser(this.federatedToken);
+        core.info("Federated token details:\n issuer - " + issuer + "\n subject claim - " + subjectClaim);
     }
 
     async validate() {
-        if (!this.servicePrincipalId || !this.tenantId || !(this.servicePrincipalKey || this.enableOIDC)) {
-            throw new Error("Not all values are present in the credentials. Ensure clientId, clientSecret and tenantId are supplied.");
+        if (!LoginConfig.azureSupportedCloudName.has(this.environment)) {
+            throw new Error("Unsupported value for environment is passed. The list of supported values for environment are 'azureusgovernment', 'azurechinacloud', 'azuregermancloud', 'azurecloud' or 'azurestack'");
         }
-        if (!this.subscriptionId && !this.allowNoSubscriptionsLogin) {
-            throw new Error("Not all values are present in the credentials. Ensure subscriptionId is supplied.");

```

Comment:
The original design makes more sense to me. Without requiring the `subscriptionId`, the user may unexpectedly perform operations in a wrong subscription.

If this feature is not related to managed identity, I suggest we split this change to another PR.